kitsune/users/tasks.py

```diff
@@ -0,0 +1,65 @@
+import json
+from datetime import datetime
+from celery import task
+from kitsune.sumo.decorators import timeit
+from kitsune.users.models import AccountEvent
+from kitsune.users.utils import anonymize_user
+from kitsune.products.models import Product
+
+
+@task
+@timeit
+def process_event_delete_user(event_id):
+    event = AccountEvent.objects.get(id=event_id)
+
+    anonymize_user(event.profile.user)
+
+    event.status = AccountEvent.PROCESSED
+    event.save()
+
+
+@task
+@timeit
+def process_event_subscription_state_change(event_id):
+    event = AccountEvent.objects.get(id=event_id)
+    body = json.loads(event.body)
+
+    last_event_q = AccountEvent.objects.filter(
+        profile_id=event.profile.pk,
+        status=AccountEvent.PROCESSED,
+        event_type=AccountEvent.SUBSCRIPTION_STATE_CHANGE
+    ).order_by("last_modified")
```

Comment:
You can also add an ordering entry in the model's `Meta` class to sort AccountEvents based on the `last_modified` event. This will apply to all events but here we are lucky because we do care about the last processed event. Then you can modify this query like `AccountEvent.objects.filter(do_some_filtering).last()`

Reply:
Ah cool, actually I realised my test wasn't resilient enough here and actually the behaviour was wrong (I needed to sort by "-last_modified" if I was going to select `last_event_q[0]`. So I've updated the test and I'll move the sorting to the model's Meta class.